dali/pipeline/data/buffer.h

```diff
@@ -218,11 +218,17 @@ class DLL_PUBLIC Buffer {
   }
 
   /**
-   * @brief Sets the type of allocation (pinned/non-pinned) for
-   * CPU buffers
+   * @brief Return true if there was data allocation
+   */
+  inline bool has_data_allocation() {
+    return !!data_;
```

Comment:
Why `!!` is needed here? Wouldn't the result be the same without it?

Reply:
`data_` is shared_ptr and it has bool-conversion defined, but `return data_` doesn't allow for implicit conversion, so I use `!!` as the idiomatic and shortest notation.
Other alternatives would be `data_.get() != nullptr` or `static_cast<bool>(data_)` which do look like a mouthful.

Reply:
I'm fine with !!, but doesn't it have a comparison with nullptr_t, so you could write `data_ != nullptr`?

Reply:
Yup, probably could also do that.